packages/shared/src/components/buttons/ButtonV2.tsx

```diff
@@ -0,0 +1,193 @@
+        !iconOnly && iconPosition === ButtonIconPosition.Left && '-ml-2 mr-1',
+        !iconOnly && iconPosition === ButtonIconPosition.Right && 'ml-1 -mr-2',
+        // add margin to social icons
+        icon.props.className?.split(' ').includes('socialIcon') && 'mr-3',
+      ),
+    });
+};
+
+export type AllowedTags = keyof Pick<JSX.IntrinsicElements, 'a' | 'button'>;
+export type AllowedElements = HTMLButtonElement | HTMLAnchorElement;
+export type ButtonElementType<Tag extends AllowedTags> = Tag extends 'a'
+  ? HTMLAnchorElement
+  : HTMLButtonElement;
+
+export type ButtonProps<T extends AllowedTags> = BaseButtonProps &
+  HTMLAttributes<T> &
+  JSX.IntrinsicElements[T] & {
+    ref?: Ref<ButtonElementType<T>>;
+  };
+
+const sizeToClassName: Record<ButtonSize, string> = {
+  [ButtonSize.XLarge]: 'border-none',
+  [ButtonSize.Large]: 'h-12 px-6 rounded-[0.88rem]',
+  [ButtonSize.Medium]: 'h-10 px-5 rounded-xl',
+  [ButtonSize.Small]: 'h-8 px-3 rounded-[0.63rem]',
+  [ButtonSize.XSmall]: 'h-6 px-2 rounded-lg',
+};
+
+const iconOnlySizeToClassName: Record<ButtonSize, string> = {
+  [ButtonSize.XLarge]: 'h-16 w-16 p-0 rounded-[1.375rem]',
+  [ButtonSize.Large]: 'h-12 w-12 p-0 rounded-[0.88rem]',
+  [ButtonSize.Medium]: 'h-10 w-10 p-0 rounded-xl',
+  [ButtonSize.Small]: 'h-8 w-8 p-0 rounded-[0.63rem]',
+  [ButtonSize.XSmall]: 'h-6 w-6 p-0 rounded-lg',
+};
+
+const variantToClassName: Record<ButtonVariant, string> = Object.values(
+  ButtonVariant,
+).reduce((acc, variant) => {
+  acc[variant] = `btn-${variant}`;
+  return acc;
+}, {} as Record<ButtonVariant, string>);
+
+const variantColorToClassName: Record<
+  ButtonVariant,
+  Record<ButtonColor, string>
+> = Object.values(ButtonVariant).reduce((variantAcc, variant) => {
+  return {
+    ...variantAcc,
+    [variant]: Object.values(ButtonColor).reduce((colorAcc, color) => {
+      return {
+        ...colorAcc,
+        [color]: `btn-${variant}-${color}`,
+      };
+    }, {} as Record<ButtonColor, string>),
+  };
+}, {} as Record<ButtonVariant, Record<ButtonColor, string>>);
+
+function ButtonComponent<TagName extends AllowedTags>(
+  {
+    variant,
+    size = ButtonSize.Medium,
+    color,
+    className,
+    icon,
+    iconPosition = ButtonIconPosition.Left,
+    loading,
+    pressed,
+    children,
+    onClick,
+    tag: Tag = 'button',
+    ...props
+  }: ButtonProps<TagName>,
+  ref?: Ref<ButtonElementType<TagName>>,
+): ReactElement {
+  const iconOnly = icon && !children;
+  const getIconWithSize = useGetIconWithSize(size, iconOnly, iconPosition);
+  const isAnchor = Tag === 'a';
+
+  return (
+    <Tag
+      {...props}
+      {...(isAnchor ? combinedClicks(onClick) : { onClick })}
+      aria-busy={loading}
+      aria-pressed={pressed}
+      ref={ref}
+      className={classNames(
+        `btn inline-flex flex-row items-center justify-center border
+        typo-callout font-bold
+        no-underline shadow-none cursor-pointer select-none focus-outline relative
```

Comment:
With the link one we add underline right in some cases? 
Just want to check it doesn't interfear

Reply:
Yes, but since the link button is currently implemented as <ClickableText> component, I think we should be safe. `no-underline` is also part of the old `<Button>` component.